tests/test_algorithm.py

```diff
@@ -1351,3 +1352,212 @@ def check_algo_positions(self, results, expected_positions):
                 actual_position = 0
 
             self.assertEqual(actual_position, expected_positions[i])
+
+
+class TestFutureChain(TestCase):
+    def setUp(self):
+        setup_logger(self)
+
+        self.metadata = {
+            0: {
+                'symbol': 'CLG06',
+                'root_symbol': 'CL',
+                'asset_type': 'future',
+                'start_date': pd.Timestamp('1980-01-01', tz='UTC'),
+                'expiration_date': pd.Timestamp('2006-01-20', tz='UTC')},
+            1: {
+                'root_symbol': 'CL',
+                'symbol': 'CLK06',
+                'asset_type': 'future',
+                'start_date': pd.Timestamp('1980-01-01', tz='UTC'),
+                'expiration_date': pd.Timestamp('2006-04-20', tz='UTC')},
+            2: {
+                'symbol': 'CLQ06',
+                'root_symbol': 'CL',
+                'asset_type': 'future',
+                'start_date': pd.Timestamp('1980-01-01', tz='UTC'),
+                'expiration_date': pd.Timestamp('2006-07-20', tz='UTC')},
+            3: {
+                'symbol': 'CLX06',
+                'root_symbol': 'CL',
+                'asset_type': 'future',
+                'start_date': pd.Timestamp('2006-02-01', tz='UTC'),
+                'expiration_date': pd.Timestamp('2006-10-20', tz='UTC')}
+        }
+
+        sids = [0, 1, 2, 3]
+
+        # Params and trade source for a 2 day simulation
+        self.sim_params_2 = factory.create_simulation_parameters(num_days=2)
+        self.source_2 = factory.create_minutely_trade_source(
+            sids,
+            sim_params=self.sim_params_2,
+            concurrent=True,
+        )
+
+        # Params and trade source for a 50 day simulation
+        self.sim_params_50 = factory.create_simulation_parameters(num_days=50)
+        self.source_50 = factory.create_minutely_trade_source(
+            sids,
+            sim_params=self.sim_params_50,
+            concurrent=True,
+        )
+
+    def tearDown(self):
+        teardown_logger(self)
+
+    def test_len(self):
```

Comment:
@ssanderson I think it makes sense to do at least some of it this way, since a lot of things are dependent on dates getting set (e.g. caching the current chain, getting the initial date from the trading algorithm). But I agree that a lot of this could be done outside of a simulation